x-pack/plugins/lens/public/app_plugin/lens_top_nav.tsx

```diff
@@ -679,32 +694,14 @@ export const LensTopNavMenu = ({
             closeDataViewEditor.current = dataViewEditor.openEditor({
               onSave: async (dataView) => {
                 if (dataView.id) {
-                  handleIndexPatternChange({
-                    activeDatasources: Object.keys(datasourceStates).reduce(
-                      (acc, datasourceId) => ({
-                        ...acc,
-                        [datasourceId]: datasourceMap[datasourceId],
-                      }),
-                      {}
-                    ),
-                    datasourceStates,
-                    indexPatternId: dataView.id,
-                    setDatasourceState,
-                  });
+                  dispatchChangeIndexPattern(dataView.id);
```

Comment:
When a new data view is created from within Lens, the data panel crashes (seems like the new data view isn't added to the cache fast enough?)
```
reportUnhandledError.js:10 Uncaught TypeError: Cannot read properties of undefined (reading 'fields')
    at InnerIndexPatternDataPanel (datapanel.tsx:211:1)
    at nh (react-dom.profiling.min.js:153:1)
    at ii (react-dom.profiling.min.js:175:1)
    at hi (react-dom.profiling.min.js:175:1)
    at fi (react-dom.profiling.min.js:174:1)
    at ek (react-dom.profiling.min.js:272:1)
    at dk (react-dom.profiling.min.js:248:1)
    at Xj (react-dom.profiling.min.js:247:1)
    at Ij (react-dom.profiling.min.js:240:1)
    at react-dom.profiling.min.js:122:1
InnerIndexPatternDataPanel	@	datapanel
```